homeassistant/components/octoprint/config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Config flow for OctoPrint integration."""
+import logging
+
+import requests
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_KEY, CONF_HOST, CONF_NAME, CONF_SENSORS
+import homeassistant.helpers.config_validation as cv
+
+from .const import (  # pylint:disable=unused-import
+    CONF_BED,
+    CONF_NUMBER_OF_TOOLS,
+    DOMAIN,
+)
+from .octoprintapi import OctoPrintAPI
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "OctoPrint"
+
+SENSORS = [
+    "Printing",
+    "Printing Error",
+    "Temperatures",
+    "Current State",
+    "Job Percentage",
+    "Time Remaining",
+    "Time Elapsed",
+]
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+        vol.Required(CONF_HOST): cv.string,
```

Comment:
It says `host`, but is an URL at this point. I would recommend it to be a hostname, port and SSL setting separate, to be more in line with how that is handled in Home Assistant in general.

Reply:
Sounds good. Split back apart.